app/src/main/java/org/mozilla/fenix/historymetadata/controller/HistoryMetadataController.kt

```diff
@@ -53,6 +64,15 @@ class DefaultHistoryMetadataController(
         )
     }
 
+    override fun handleRemoveGroup(searchTerm: String) {
+        scope.launch {
+            storage.deleteHistoryMetadata(searchTerm)
+        }
+        navController.navigate(
+            BrowserFragmentDirections.actionGlobalHome()
```

Comment:
That's true 🤔 Maybe something like this? --

```
scope.launch {
            storage.deleteHistoryMetadata(searchTerm)
        }.invokeOnCompletion {
            navController.navigate(
                BrowserFragmentDirections.actionGlobalHome()
            )
        }
```

Reply:
I'm not sure what the best way to block on the deletion is

Reply:
This is fine for a follow-up.

Semantics of `invokeOnCompletion` aren't very clear to me after reading its docs - e.g. which thread will the completion block be called in?

I think just a simple thing like this would work:
```
    scope.launch {
            withContext(Dispatchers.IO) {
                storage.deleteHistoryMetadata(searchTerm)
            }

            navController.navigate(
                BrowserFragmentDirections.actionGlobalHome()
            )
        }
```